src/charm.py

```diff
@@ -43,17 +51,19 @@ class MimirCoordinatorK8SOperatorCharm(ops.CharmBase):
     def __init__(self, *args: Any):
         super().__init__(*args)
 
+        self._nginx_container = self.unit.get_container("nginx")
+        self._nginx_prometheus_exporter_container = self.unit.get_container(
+            "nginx-prometheus-exporter"
+        )
```

Comment:
I'm not sure it's a good idea, because this is how we do it in all of our charms (e.g. [tempo-coordinator](https://github.com/canonical/tempo-coordinator-k8s-operator/blob/main/src/charm.py#L156) but also [prometheus](https://github.com/canonical/prometheus-k8s-operator/blob/main/src/charm.py#L152-L154), [loki](https://github.com/canonical/loki-k8s-operator/blob/main/src/charm.py#L140-L141), etc)